.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+# Build the package and run tests, using the latest trifinger_user Apptainer
+# image.
+# The Apptainer image has a full ROBOT_FINGERS workspace installed, so all
+# dependencies are there and only the checked package needs to be build, not the
+# whole workspace.
+#
+# Note: The Apptainer image only gets automatically rebuild, if something in the
+# image definition changes, not when other packages are changed.  So to avoid
+# that the workspace in the container gets outdated, manual builds need to be
+# triggered regularly.
+
+name: Build and Test
+
+on:
+  pull_request:
+  # NOTE: By also running on master (=the default branch), we have a cache for
+  # the default branch which will be used by other branches if they don't have
+  # their own cache already (see scope of caches created by actions/cache).
+  push:
+    branches:
+     - master
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-20.04
+    steps:
+      # write the current week of the year to an output parameter so it can
+      # later be used in the cache key
+      - name: Get current week of the year
+        id: date
+        run: echo "::set-output name=week::$(date +'%W')"
+
+      # Cache the .apptainer/ directory, so the image does not need to be
+      # downloaded every time.
+      # The cache is created if it does not yet exists but changes in the cache
+      # are not stored in later runs.  As this makes the cache useless once
+      # the image changed, add the number of the current week to its key.  This
+      # way a new cache will be created once per week.
+      - name: Apptainer Cache
+        id: cache-apptainer
+        uses: actions/cache@v3
+        with:
+          path: ~/.apptainer/cache
+          key: ${{ runner.os }}-apptainer-${{ steps.date.outputs.week }}
+
+      - name: Install Apptainer
```

Comment:
I find caching very confusing. 

Is it because we "tagged" ```~/.apptainer/cache``` as cache (in previous job) that it will not download the deb file again ? (without the need to explicitely ask to use the cache if availalbe) but apt-get install will occur in everycase ?

Or does the "cache job" triggers the cache to be used always when possible (e..g possibly apt-get install of the deb file is not run again)

Reply:
note: if that is something you will present for the software tutorial, I can wait for the answer :)

Reply:
I'm wondering if it is maybe overkill here (the deb file is only 27 MB) and makes this step a bit too complicated.

Reply:
to be sure I understood: the job "Apptainer Cache" simply say not to delete the folder ~/.apptainer/cache and the files ~/apptainer_*_amd64.deb ? (and nothing else ?)

Reply:
More or less.  At the end of the first run, it saves them and restores them in following runs.  However, it will always use the snapshot from the first run, i.e. updates from following runs are not preserved.
That's why I add the week number, so once per week the cache is renewed.